libs/execution/include/hpx/execution/this_thread.hpp

```diff
@@ -0,0 +1,53 @@
+//  Copyright (c) 2019 Thomas Heller
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef HPX_EXECUTION_THIS_THREAD_HPP
+#define HPX_EXECUTION_THIS_THREAD_HPP
+
+#include <hpx/execution/execution_context.hpp>
+#include <hpx/timing/steady_clock.hpp>
+
+#include <cstdint>
+
+namespace hpx { namespace execution { namespace this_thread {
+    namespace detail {
+        struct execution_context_storage;
+        execution_context_storage* get_execution_context_storage();
+    }    // namespace detail
+
+    struct reset_execution_context
+    {
+        reset_execution_context(
+            detail::execution_context_storage*, execution_context_base& impl);
+        reset_execution_context(execution_context_base& impl);
+        ~reset_execution_context();
+
+        detail::execution_context_storage* storage_;
+        execution_context_base* old_;
+    };
+
+    hpx::execution::execution_context execution_context();
+
+    void yield(const char* desc = "hpx::execution::this_thread::yield");
+    void yield_k(std::size_t k,
+        const char* desc = "hpx::execution::this_thread::yield_k");
+    void suspend(const char* desc = "hpx::execution::this_thread::suspend");
```

Comment:
Will we have to `HPX_EXPORT` these functions?

Reply:
Right. All symbols have been exported properly now.